docs/mp/testing.adoc

```diff
@@ -36,25 +47,77 @@ include::{rootdir}/includes/dependencies.adoc[]
+
+*If you want to use method sources that use CDI with repeatable tests, please do not use `@Configuration(useExisting=true)`*
+
+=== Usage - added parameters and injection types
+The following types are available for injection (when a single CDI container is used per test class):
+
+- `WebTarget` - a JAX-RS client's target configured for the current hostname and port when `helidon-micorprofile-server` is on
+the classpath
+
+The following types are available as method parameters (in any type of Helidon tests):
+
+- `WebTarget` - a JAX-RS client's target configured for the current hostname and port when `helidon-micorprofile-server` is on
+the classpath
+- `SeContainer` - the current container instance
+
+== API
+
 The annotations described in this section are inherited (for the non-repeatable ones), and additive (for repeatable).
 So if you declare `@DisableDiscovery` on abstract class, all implementations will have discovery disabled, unless you
 annotate the implementation class with `@DisableDiscovery(false)`.
 If you declare `@AddBean` on both abstract class and implementation class, both beans will be added.
 
 In addition to this simplification, the following annotations are supported:
 
-- `io.helidon.microprofile.tests.junit5.AddBean` - to add one or more beans to the container
-        (if not part of a bean archive, or when discovery is disabled)
-- `io.helidon.microprofile.tests.junit5.AddExtension` - to add one or more CDI extensions to the container
-        (if not added through service loader, or when discovery is disabled)
-- `io.helidon.microprofile.tests.junit5.AddConfig` - to add one or more configuration properties to MicroProfile config
-        without the need of creating a `microprofile-config.properties` file
-- `io.helidon.microprofile.tests.junit5.DisableDiscovery` - to disable automated discovery of beans and extensions
+|===
+|Annotation | Usage
+
+|`@io.helidon.microprofile.tests.junit5.AddBean`
+|Used to add one or more beans to the container (if not part of a bean archive, or when discovery is disabled)
+
+|`@io.helidon.microprofile.tests.junit5.AddExtension`
+|Used to add one or more CDI extensions to the container (if not added through service loader, or when discovery is disabled)
+
+|`@io.helidon.microprofile.tests.junit5.AddConfig`
+|Used to add one or more configuration properties to MicroProfile config without the need of creating a `microprofile-config.properties` file
+
+|Used `@io.helidon.microprofile.tests.junit5.DisableDiscovery`
+|to disable automated discovery of beans and extensions
+|===
+
+== Examples
+
+In current example Helidon container will be launched prior test. The _Bean Discovery_ will be disabled. _MyBean_ will be added to the test, so that it can be injected. _ConfigCdiExtension_ will be enabled for this test. And finally, a configuration property will be added using `@AddConfig` annotation.
```

Comment:
In this example, the Helidon container will be launched before the test.

Reply:
fixed